### PR TITLE
Add support for floating-point guifont height

### PR DESCRIPTION
**Pull request by smolck**
This works as intended for me, let me know what you think.

See #65.

### CODE REVIEW

**Comment by badosu** [comment]
Works for me!

Some tests are failing though, current one because previously font size was unsigned and float accepts negative values.

I think it's acceptable to ignore negative values as we are doing right now, but if previous behavior is wanted you can split the condition on https://github.com/smolck/gnvim/blob/9ea48e92fab0730b1530ce4145fb76be49332635/src/ui/font.rs#L51 and return `Err()` when negative.

**Comment by last-partizan** [comment]
I just compiled it from your branch and it works like expected. Thanks!

One more point, how do i set guifont in config? does gnvim have something like `.gnvimrc`? or `has('gui_gnvim`)`?

**Comment by smolck** [comment]
@last-partizan This should work, since GNvim sets `g:gnvim` to 1 when it is running:

```
if exists(‘g:gnvim’)
    set guifont=<your font and height>
    “ Other settings for GNvim
endif
```

This isn’t strictly necessary, since doing a simple `set guifont=...` in your `init.vim`/`.vimrc` shouldn’t break anything and would work as you want (I think).

**Comment by smolck** [comment]
@badosu Nice spot! Didn't notice that. How about this?
```diff
diff --git a/src/ui/font.rs b/src/ui/font.rs
index 4d4ecf2..9bc5cd6 100644
--- a/src/ui/font.rs
+++ b/src/ui/font.rs
@@ -48,6 +48,7 @@ impl Font {
                     'h' => {
                         let rest = chars.collect::<String>();
                         let h = rest.parse::<f32>().or(Err(()))?;
+
                         if h == 0.0 || h < 0.0 {
                             // Ignore zero sized font.
                             continue;
@@ -146,15 +147,18 @@ mod tests {
         assert_eq!(f.is_err(), true);
         let f = Font::from_guifont("font:hn");
         assert_eq!(f.is_err(), true);
-        let f = Font::from_guifont("font:h-1");
-        assert_eq!(f.is_err(), true);
 
         // Font with height zero.
         let f = Font::from_guifont("foo:h0").unwrap();
         assert_eq!(f.name, "foo");
         assert_eq!(f.height, DEFAULT_HEIGHT);
 
-        // Font with no hegith.
+        // Font with negative height.
+        let f = Font::from_guifont("font:h-1").unwrap();
+        assert_eq!(f.name, "font");
+        assert_eq!(f.height, DEFAULT_HEIGHT);
+
+        // Font with no height.
         let f = Font::from_guifont("bar").unwrap();
         assert_eq!(f.name, "bar");
         assert_eq!(f.height, DEFAULT_HEIGHT);
```

I think ignoring the negative font is a better option. If someone types a negative number for height, I think changing their font type might be annoying. It'd be nice to set the font size to default to the current font size (if it is set) when an invalid font size is given, but that's a matter for another PR (and a different file).

**Comment by badosu** [comment]
> I think ignoring the case of a negative font is better than returning an error

I agree with this approach :+1:.

File: src/ui/font.rs [hunk 1]
```diff
@@ -139,22 +139,25 @@ mod tests {
...
-        assert_eq!(f.is_err(), true);
 
         // Font with height zero.
         let f = Font::from_guifont("foo:h0").unwrap();
         assert_eq!(f.name, "foo");
         assert_eq!(f.height, DEFAULT_HEIGHT);
 
-        // Font with no hegith.
+        // Font with negative height.
+        let f = Font::from_guifont("font:h-1");
```
Comment (badosu) [inline src/ui/font.rs]:
```suggestion
         let f = Font::from_guifont("font:h-1").unwrap()
```

Comment (smolck) [reply src/ui/font.rs]: Thanks @badosu! Didn't see that!

File: src/ui/font.rs [hunk 2]
```diff
@@ -47,8 +47,8 @@ impl Font {
                 match ch {
                     'h' => {
                         let rest = chars.collect::<String>();
-                        let h = rest.parse::<usize>().or(Err(()))?;
-                        if h == 0 {
+                        let h = rest.parse::<f32>().or(Err(()))?;
+                        if h == 0.0 || h < 0.0 {
```
Comment (vhakulinen) [inline src/ui/font.rs]: `h <= 0.0` is enough.

Comment (smolck) [reply src/ui/font.rs]: @vhakulinen Thank you! Should be fixed now.

**Pull request merged**